// igniteui-react-maps/esm5/lib/GeographicMapSeriesHost_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { Series } from "igniteui-react-charts";
import { DependencyProperty } from "igniteui-react-core";
import { typeCast, Number_$type, runOn, delegateRemove, delegateCombine, markType, TypeRegistrar } from "igniteui-react-core";
import { SeriesView } from "igniteui-react-charts";
import { SeriesViewer } from "igniteui-react-charts";
import { PropertyMetadata } from "igniteui-react-core";
import { PlatformConstants } from "igniteui-react-core";
import { isNaN_ } from "igniteui-react-core";
/**
 * @hidden
 */
var GeographicMapSeriesHost = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicMapSeriesHost, _super);
    function GeographicMapSeriesHost() {
        var _this = _super.call(this) || this;
        _this.l1 = null;
        _this.l3 = _this.l2();
        return _this;
    }
    GeographicMapSeriesHost.prototype.get_ct = function () {
        return true;
    };
    GeographicMapSeriesHost.prototype.fs = function (a, b) {
        var c = this.gv(b);
        if (c == "LatitudeMemberPath") {
            c = "YMemberPath";
        }
        if (c == "LongitudeMemberPath") {
            c = "XMemberPath";
        }
        return this.l3.fs(a, b);
    };
    GeographicMapSeriesHost.prototype.ch = function (a, b, c) {
        return this.l3.ch(a, b, c);
    };
    GeographicMapSeriesHost.prototype.ag = function () {
        return this.l3.ag();
    };
    GeographicMapSeriesHost.prototype.a5 = function () {
        return this.l3;
    };
    Object.defineProperty(GeographicMapSeriesHost.prototype, "l3", {
        get: function () {
            return this.l1;
        },
        set: function (a) {
            var b = this.l3 != a;
            if (b) {
                this.l1 = a;
                this.l8();
            }
        },
        enumerable: true,
        configurable: true
    });
    GeographicMapSeriesHost.prototype.l8 = function () {
        if (this.l1 != null) {
            var hostedSeries_ = this.l1;
            hostedSeries_.externalObject = TypeRegistrar.createFromInternal(hostedSeries_, PlatformConstants.Prefix, PlatformConstants.Postfix);
            hostedSeries_.externalObject._implementation = hostedSeries_;
        }
        this.h1(this.l3);
        this.l4.bv();
    };
    GeographicMapSeriesHost.prototype.fp = function (a) {
        return this.fu(this.l3, a);
    };
    GeographicMapSeriesHost.prototype.bf = function () {
        return new HostSeriesView(this);
    };
    GeographicMapSeriesHost.prototype.iv = function (a) {
        _super.prototype.iv.call(this, a);
        this.l4 = typeCast(HostSeriesView.$, a);
    };
    Object.defineProperty(GeographicMapSeriesHost.prototype, "l5", {
        get: function () {
            return this.c(GeographicMapSeriesHost.ma);
        },
        set: function (a) {
            this.h(GeographicMapSeriesHost.ma, a);
        },
        enumerable: true,
        configurable: true
    });
    GeographicMapSeriesHost.prototype.i0 = function (a, b, c, d) {
        _super.prototype.i0.call(this, a, b, c, d);
        switch (b) {
            case "SeriesViewer":
                this.l3.bs = this.bs;
                if (c != null) {
                    c.jy(this.l3);
                }
                if (d != null) {
                    this.bs.hc(this.l3);
                }
                this.l3.i3(this.lz);
                this.l3.fh = this.fh;
                this.h6(this.l3);
                this.fk = this.l3.fk;
                var e = typeCast(SeriesViewer.$, c);
                if (e != null) {
                    e.actualWindowRectChanged = delegateRemove(e.actualWindowRectChanged, runOn(this, this.l9));
                }
                if (this.bs != null) {
                    var f = this.bs;
                    f.actualWindowRectChanged = delegateCombine(f.actualWindowRectChanged, runOn(this, this.l9));
                }
                this.l7();
                break;
            case "SyncLink":
                this.l3.bn = this.bn;
                break;
            case "VisibleFromScale":
                this.l7();
                break;
            case "Index":
                this.l3.fh = this.fh;
                break;
            case "ItemsSource":
                this.l4.bw();
                break;
            case "Resolution":
                this.l4.by();
                break;
            case "TransitionDuration":
                this.l4.bz();
                break;
            case "Opacity":
                this.l4.bx();
                break;
            case "Visibility":
                var hs_ = this.l3;
                var v_ = this._visibility;
                hs_.visibility = v_;
                break;
            case "MouseOverEnabled":
                this.l3.c7 = this.c7;
                break;
            case "ExpectFunctions":
                this.l3.expectFunctions = this.expectFunctions;
                break;
        }
    };
    GeographicMapSeriesHost.prototype.l7 = function () {
        if (this.bs == null || this.l3 == null) {
            return;
        }
        if (!isNaN_(this.l5)) {
            if (this.bs.mr.width > this.l5) {
                var hs_ = this.l3;
                var v_ = 1;
                hs_.visibility = v_;
            }
            else {
                var hs2_ = this.l3;
                var v2_ = 0;
                hs2_.visibility = v2_;
            }
        }
    };
    GeographicMapSeriesHost.prototype.l9 = function (a, b) {
        this.l7();
    };
    GeographicMapSeriesHost.prototype.je = function (a) {
        _super.prototype.je.call(this, a);
        if (this.l3 != null) {
            this.l3.je(a);
        }
    };
    GeographicMapSeriesHost.prototype.jd = function (a, b, c, d, e) {
        if (this.l3 != null) {
            this.l3.jd(a, b, c, d, e);
        }
    };
    GeographicMapSeriesHost.prototype.ar = function (a, b) {
        return this.l3.ar(a, b);
    };
    GeographicMapSeriesHost.prototype.jn = function () {
        _super.prototype.jn.call(this);
        this.l3.jn();
    };
    GeographicMapSeriesHost.prototype.get_coercionMethods = function () {
        return this._coercionMethods;
    };
    GeographicMapSeriesHost.prototype.set_coercionMethods = function (a) {
        this._coercionMethods = a;
        this.l3.coercionMethods = a;
    };
    GeographicMapSeriesHost.prototype.h5 = function (a) {
        _super.prototype.h5.call(this, a);
        if (this.l3 != null) {
            var b = this.l3.fo();
            this.hy(a, b);
        }
    };
    GeographicMapSeriesHost.$t = markType(GeographicMapSeriesHost, 'GeographicMapSeriesHost', Series.$);
    GeographicMapSeriesHost.ma = DependencyProperty.i("VisibleFromScale", Number_$type, GeographicMapSeriesHost.$, new PropertyMetadata(2, 1, function (a, b) {
        var c = a;
        c.raisePropertyChanged("VisibleFromScale", b.oldValue, b.newValue);
    }));
    return GeographicMapSeriesHost;
}(Series));
export { GeographicMapSeriesHost };
/**
 * @hidden
 */
var HostSeriesView = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(HostSeriesView, _super);
    function HostSeriesView(a) {
        var _this = _super.call(this, a) || this;
        _this.bu = null;
        _this.bu = a;
        return _this;
    }
    HostSeriesView.prototype.ao = function (a, b) {
        _super.prototype.ao.call(this, a, b);
    };
    HostSeriesView.prototype.bv = function () {
        this.bu.l3.itemsSource = this.bu.itemsSource;
        this.bu.l3.fh = this.bu.fh;
        this.bu.l3.k6 = this.bu.k6;
        this.bu.l3.k9 = this.bu.k9;
        this.bu.l3.eu = this.bu.eu;
        this.bu.l3.fi = this.bu.fi;
        this.bu.l3._opacity = this.bu._opacity;
    };
    HostSeriesView.prototype.bw = function () {
        this.bu.l3.itemsSource = this.bu.itemsSource;
    };
    HostSeriesView.prototype.by = function () {
        this.bu.l3.eu = this.bu.eu;
    };
    HostSeriesView.prototype.bz = function () {
        this.bu.l3.fi = this.bu.fi;
    };
    HostSeriesView.prototype.bx = function () {
        this.bu.l3._opacity = this.bu._opacity;
    };
    HostSeriesView.$t = markType(HostSeriesView, 'HostSeriesView', SeriesView.$);
    return HostSeriesView;
}(SeriesView));
export { HostSeriesView };
