// igniteui-react-maps/esm5/lib/GeographicShapeSeriesView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { GeographicShapeSeriesBaseView } from "./GeographicShapeSeriesBaseView";
import { GeographicShapeSeriesBase } from "./GeographicShapeSeriesBase";
import { ScatterPolygonSeries } from "igniteui-react-charts";
import { markType, runOn, delegateCombine, enumGetBox, markDep } from "igniteui-react-core";
import { StyleSelector } from "igniteui-react-charts";
import { DependencyProperty } from "igniteui-react-core";
import { Style } from "igniteui-react-core";
import { MarkerType_$type } from "igniteui-react-charts";
import { DataTemplate } from "igniteui-react-core";
import { Brush } from "igniteui-react-core";
import { CollisionAvoidanceType_$type } from "igniteui-react-charts";
import { PropertyMetadata } from "igniteui-react-core";
import { DeviceUtils } from "igniteui-react-core";
/**
 * @hidden
 */
var GeographicShapeSeriesView = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicShapeSeriesView, _super);
    function GeographicShapeSeriesView(a) {
        var _this = _super.call(this, a) || this;
        _this.b7 = null;
        _this.b7 = a;
        if (!_this.n) {
            _this.b7.mi = "points";
        }
        return _this;
    }
    GeographicShapeSeriesView.prototype.bv = function () {
        _super.prototype.bv.call(this);
        this.b7.l3.ok = this.b7.nh;
        this.b7.l3.nk = this.b7.mq;
        this.b7.l3.n1 = this.b7.m4;
        this.b7.l3.nm = this.b7.ms;
        this.b7.l3.ne = this.b7.mo;
    };
    GeographicShapeSeriesView.prototype.cc = function () {
        this.b7.l3.n1 = this.b7.m4;
    };
    GeographicShapeSeriesView.prototype.ce = function () {
        this.b7.l3.nm = this.b7.ms;
    };
    GeographicShapeSeriesView.prototype.b9 = function () {
        this.b7.l3.ne = this.b7.mo;
    };
    GeographicShapeSeriesView.prototype.cd = function () {
        this.b7.l3.nf = this.b7.mp;
    };
    GeographicShapeSeriesView.prototype.b8 = function () {
        this.b7.l3.og = this.b7.ne;
    };
    GeographicShapeSeriesView.prototype.ca = function () {
        this.b7.l3.oh = this.b7.nf;
    };
    GeographicShapeSeriesView.prototype.cb = function () {
        this.b7.l3.oj = this.b7.ng;
    };
    GeographicShapeSeriesView.prototype.cf = function () {
        this.b7.l3.nk = this.b7.mq;
    };
    GeographicShapeSeriesView.prototype.cg = function () {
        this.b7.l3.ok = this.b7.nh;
    };
    GeographicShapeSeriesView.$t = markType(GeographicShapeSeriesView, 'GeographicShapeSeriesView', GeographicShapeSeriesBaseView.$);
    return GeographicShapeSeriesView;
}(GeographicShapeSeriesBaseView));
export { GeographicShapeSeriesView };
/**
 * @hidden
 */
var GeographicShapeSeries = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicShapeSeries, _super);
    function GeographicShapeSeries() {
        var _this = _super.call(this) || this;
        _this.styleShape = null;
        _this.ab = GeographicShapeSeries.$;
        return _this;
    }
    GeographicShapeSeries.prototype.bf = function () {
        return new GeographicShapeSeriesView(this);
    };
    GeographicShapeSeries.prototype.iv = function (a) {
        _super.prototype.iv.call(this, a);
        this.mr = a;
    };
    GeographicShapeSeries.prototype.l2 = function () {
        var a = new ScatterPolygonSeries();
        a.styleShape = delegateCombine(a.styleShape, runOn(this, this.m3));
        return a;
    };
    GeographicShapeSeries.prototype.m3 = function (a, b) {
        if (this.styleShape != null) {
            this.styleShape(this, b);
        }
    };
    GeographicShapeSeries.prototype.i0 = function (a, b, c, d) {
        _super.prototype.i0.call(this, a, b, c, d);
        switch (b) {
            case GeographicShapeSeries.$$p[4]:
                this.mr.cc();
                break;
            case GeographicShapeSeries.$$p[5]:
                this.mr.ce();
                break;
            case GeographicShapeSeries.$$p[1]:
                this.mr.b9();
                break;
            case GeographicShapeSeries.$$p[6]:
                this.mr.cd();
                break;
            case GeographicShapeSeries.$$p[0]:
                this.mr.b8();
                break;
            case GeographicShapeSeries.$$p[2]:
                this.mr.ca();
                break;
            case GeographicShapeSeries.$$p[3]:
                this.mr.cb();
                break;
            case GeographicShapeSeries.$$p[8]:
                this.mr.cf();
                break;
            case GeographicShapeSeries.$$p[7]:
                this.mr.cg();
                break;
        }
    };
    GeographicShapeSeries.$t = markType(GeographicShapeSeries, 'GeographicShapeSeries', GeographicShapeSeriesBase.$);
    GeographicShapeSeries.$$p = markDep(DependencyProperty, PropertyMetadata, GeographicShapeSeries, 'raisePropertyChanged', ['MarkerBrush:ne:m5', [Brush.$], 'MarkerCollisionAvoidance:mo:m6', [CollisionAvoidanceType_$type, enumGetBox(CollisionAvoidanceType_$type, 0)], 'MarkerOutline:nf:m7', [Brush.$], 'MarkerStyle:ng:m8', [Style.$], 'MarkerTemplate:m4:m9', [DataTemplate.$, null], 'MarkerThickness:ms:na', [1, DeviceUtils.g(2)], 'MarkerType:mp:nb', [MarkerType_$type, enumGetBox(MarkerType_$type, 1)], 'ShapeStyle:nh:nc', [Style.$, null], 'ShapeStyleSelector:mq:nd', [StyleSelector.$, null]]);
    return GeographicShapeSeries;
}(GeographicShapeSeriesBase));
export { GeographicShapeSeries };
