// igniteui-react-maps/esm5/lib/GeographicProportionalSymbolSeriesView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { HostSeriesView } from "./HostSeriesView";
import { GeographicMapSeriesHost } from "./GeographicMapSeriesHost";
import { BubbleSeries } from "igniteui-react-charts";
import { markType, enumGetBox, markDep } from "igniteui-react-core";
import { DependencyProperty } from "igniteui-react-core";
import { MarkerType_$type } from "igniteui-react-charts";
import { DataTemplate } from "igniteui-react-core";
import { Brush } from "igniteui-react-core";
import { SizeScale } from "igniteui-react-charts";
import { BrushScale } from "igniteui-react-charts";
import { PropertyMetadata } from "igniteui-react-core";
import { DeviceUtils } from "igniteui-react-core";
/**
 * @hidden
 */
var GeographicProportionalSymbolSeriesView = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicProportionalSymbolSeriesView, _super);
    function GeographicProportionalSymbolSeriesView(a) {
        var _this = _super.call(this, a) || this;
        _this.b0 = null;
        _this.b0 = a;
        return _this;
    }
    GeographicProportionalSymbolSeriesView.prototype.bv = function () {
        _super.prototype.bv.call(this);
        var a = this.b0.l3;
        a.nv = this.b0.mv;
        a.nz = this.b0.mt;
        a.l2 = this.b0.md;
        a.mn = this.b0.na;
        a.l7 = this.b0.ml;
        a.px = this.b0.m6;
        a.o5 = this.b0.me;
        a.pr = this.b0.mr;
        a.pm = this.b0.mn;
        a.o1 = this.b0.mb;
        a.pd = this.b0.mg;
        a.pe = this.b0.mh;
        a.pk = this.b0.mk;
        a.pj = this.b0.mj;
    };
    GeographicProportionalSymbolSeriesView.prototype.b9 = function () {
        this.b0.l3.nv = this.b0.mv;
    };
    GeographicProportionalSymbolSeriesView.prototype.b8 = function () {
        this.b0.l3.nz = this.b0.mt;
    };
    GeographicProportionalSymbolSeriesView.prototype.ce = function () {
        this.b0.l3.l2 = this.b0.md;
    };
    GeographicProportionalSymbolSeriesView.prototype.cc = function () {
        this.b0.l3.mn = this.b0.na;
    };
    GeographicProportionalSymbolSeriesView.prototype.cd = function () {
        this.b0.l3.l7 = this.b0.ml;
    };
    GeographicProportionalSymbolSeriesView.prototype.cf = function () {
        this.b0.l3.no = this.b0.mm;
    };
    GeographicProportionalSymbolSeriesView.prototype.ch = function () {
        this.b0.l3.xAxis = this.b0.bs.nc;
    };
    GeographicProportionalSymbolSeriesView.prototype.ci = function () {
        this.b0.l3.yAxis = this.b0.bs.ne;
    };
    GeographicProportionalSymbolSeriesView.prototype.cg = function () {
        var a = this.b0.l3;
        if (this.b0.bs == null) {
            a.xAxis = null;
            a.yAxis = null;
            return;
        }
        a.xAxis = this.b0.bs.nc;
        a.yAxis = this.b0.bs.ne;
    };
    GeographicProportionalSymbolSeriesView.prototype.ca = function () {
        this.b0.l3.m1 = this.b0.nt;
    };
    GeographicProportionalSymbolSeriesView.prototype.cb = function () {
        this.b0.l3.m2 = this.b0.nu;
    };
    GeographicProportionalSymbolSeriesView.prototype.cj = function () {
        this.b0.l3.px = this.b0.m6;
    };
    GeographicProportionalSymbolSeriesView.prototype.ck = function () {
        this.b0.l3.o5 = this.b0.me;
        this.b0.l3.pf = this.b0.mi;
    };
    GeographicProportionalSymbolSeriesView.prototype.b4 = function () {
        this.b0.l3.pr = this.b0.mr;
    };
    GeographicProportionalSymbolSeriesView.prototype.b2 = function () {
        this.b0.l3.o1 = this.b0.mb;
        this.b0.l3.pd = this.b0.mg;
        this.b0.l3.pe = this.b0.mh;
        this.b0.l3.pk = this.b0.mk;
        this.b0.l3.pj = this.b0.mj;
    };
    GeographicProportionalSymbolSeriesView.prototype.b1 = function () {
        this.b0.l3.pm = this.b0.mn;
    };
    GeographicProportionalSymbolSeriesView.prototype.cl = function () {
        this.b0.l3.pf = this.b0.mi;
    };
    GeographicProportionalSymbolSeriesView.prototype.b3 = function () {
        this.b0.l3.pd = this.b0.mg;
    };
    GeographicProportionalSymbolSeriesView.prototype.b7 = function () {
        this.b0.l3.pe = this.b0.mh;
    };
    GeographicProportionalSymbolSeriesView.prototype.b6 = function () {
        this.b0.l3.pk = this.b0.mk;
    };
    GeographicProportionalSymbolSeriesView.prototype.b5 = function () {
        this.b0.l3.pj = this.b0.mj;
    };
    GeographicProportionalSymbolSeriesView.$t = markType(GeographicProportionalSymbolSeriesView, 'GeographicProportionalSymbolSeriesView', HostSeriesView.$);
    return GeographicProportionalSymbolSeriesView;
}(HostSeriesView));
export { GeographicProportionalSymbolSeriesView };
/**
 * @hidden
 */
var GeographicProportionalSymbolSeries = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicProportionalSymbolSeries, _super);
    function GeographicProportionalSymbolSeries() {
        var _this = _super.call(this) || this;
        _this.mc = null;
        _this.mh = false;
        _this.mk = NaN;
        _this.mj = NaN;
        _this.ab = GeographicProportionalSymbolSeries.$;
        return _this;
    }
    GeographicProportionalSymbolSeries.prototype.bf = function () {
        return new GeographicProportionalSymbolSeriesView(this);
    };
    GeographicProportionalSymbolSeries.prototype.iv = function (a) {
        _super.prototype.iv.call(this, a);
        this.mf = a;
    };
    GeographicProportionalSymbolSeries.prototype.l2 = function () {
        return new BubbleSeries();
    };
    GeographicProportionalSymbolSeries.prototype.i0 = function (a, b, c, d) {
        _super.prototype.i0.call(this, a, b, c, d);
        switch (b) {
            case "SeriesViewer":
                this.mf.cg();
                break;
            case GeographicProportionalSymbolSeries.$$p[5]:
                this.mf.b9();
                break;
            case GeographicProportionalSymbolSeries.$$p[4]:
                this.mf.b8();
                break;
            case GeographicProportionalSymbolSeries.$$p[13]:
                this.mf.ce();
                break;
            case GeographicProportionalSymbolSeries.$$p[11]:
                this.mf.cc();
                break;
            case GeographicProportionalSymbolSeries.$$p[12]:
                this.mf.cd();
                break;
            case GeographicProportionalSymbolSeries.$$p[14]:
                this.mf.cf();
                break;
            case "XAxis":
                this.mf.ch();
                break;
            case "YAxis":
                this.mf.ci();
                break;
            case GeographicProportionalSymbolSeries.$$p[7]:
                this.mf.ca();
                break;
            case GeographicProportionalSymbolSeries.$$p[9]:
                this.mf.cb();
                break;
            case GeographicProportionalSymbolSeries.$$p[15]:
                this.mf.cj();
                break;
            case GeographicProportionalSymbolSeries.$$p[16]:
                this.mf.ck();
                break;
            case GeographicProportionalSymbolSeries.$$p[17]:
                this.mf.cl();
                break;
            case GeographicProportionalSymbolSeries.$$p[3]:
                this.mf.b4();
                break;
            case GeographicProportionalSymbolSeries.$$p[0]:
                this.mf.b1();
                break;
            case GeographicProportionalSymbolSeries.$$p[1]:
                this.mf.b2();
                break;
            case GeographicProportionalSymbolSeries.$$p[2]:
                this.mf.b3();
                break;
            case GeographicProportionalSymbolSeries.$$p[10]:
                this.mf.b7();
                break;
            case GeographicProportionalSymbolSeries.$$p[8]:
                this.mf.b6();
                break;
            case GeographicProportionalSymbolSeries.$$p[6]:
                this.mf.b5();
                break;
        }
    };
    GeographicProportionalSymbolSeries.$t = markType(GeographicProportionalSymbolSeries, 'GeographicProportionalSymbolSeries', GeographicMapSeriesHost.$);
    GeographicProportionalSymbolSeries.$$p = markDep(DependencyProperty, PropertyMetadata, GeographicProportionalSymbolSeries, 'raisePropertyChanged', ['FillMemberPath:mn:nb', [2], 'FillScale:mb:nc', [BrushScale.$], 'FillScaleUseGlobalValues:mg:nd', [0], 'LabelMemberPath:mr:ne', [2], 'LatitudeMemberPath:mt:nf', [2, null], 'LongitudeMemberPath:mv:ng', [2, null], 'MarkerBrushBrightness:mj:nh', [1], 'MarkerBrush:nt:ni', [Brush.$], 'MarkerOutlineBrightness:mk:nj', [1], 'MarkerOutline:nu:nk', [Brush.$], 'MarkerOutlineUsesFillScale:mh:nl', [0], 'MarkerTemplate:na:nm', [DataTemplate.$, null], 'MarkerThickness:ml:nn', [1, DeviceUtils.g(2)], 'MarkerType:md:no', [MarkerType_$type, enumGetBox(MarkerType_$type, 1)], 'MaximumMarkers:mm:np', [1, 400], 'RadiusMemberPath:m6:nq', [2, null], 'RadiusScale:me:nr', [SizeScale.$], 'RadiusScaleUseGlobalValues:mi:ns', [0]]);
    return GeographicProportionalSymbolSeries;
}(GeographicMapSeriesHost));
export { GeographicProportionalSymbolSeries };
