// igniteui-react-maps/esm5/lib/GeographicPolylineSeriesView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { GeographicShapeSeriesBaseView } from "./GeographicShapeSeriesBaseView";
import { ScatterPolylineSeries } from "igniteui-react-charts";
import { markType, runOn, delegateCombine, markDep } from "igniteui-react-core";
import { GeographicShapeSeriesBase } from "./GeographicShapeSeriesBase";
import { StyleSelector } from "igniteui-react-charts";
import { DependencyProperty } from "igniteui-react-core";
import { Style } from "igniteui-react-core";
import { PropertyMetadata } from "igniteui-react-core";
/**
 * @hidden
 */
var GeographicPolylineSeriesView = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicPolylineSeriesView, _super);
    function GeographicPolylineSeriesView(a) {
        var _this = _super.call(this, a) || this;
        _this.b7 = null;
        _this.b7 = a;
        return _this;
    }
    GeographicPolylineSeriesView.prototype.bv = function () {
        _super.prototype.bv.call(this);
        this.b7.l3.nd = this.b7.mo;
        this.b7.l3.ni = this.b7.mw;
    };
    GeographicPolylineSeriesView.prototype.b8 = function () {
        this.b7.l3.nd = this.b7.mo;
    };
    GeographicPolylineSeriesView.prototype.b9 = function () {
        this.b7.l3.ni = this.b7.mw;
    };
    GeographicPolylineSeriesView.$t = markType(GeographicPolylineSeriesView, 'GeographicPolylineSeriesView', GeographicShapeSeriesBaseView.$);
    return GeographicPolylineSeriesView;
}(GeographicShapeSeriesBaseView));
export { GeographicPolylineSeriesView };
/**
 * @hidden
 */
var GeographicPolylineSeries = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicPolylineSeries, _super);
    function GeographicPolylineSeries() {
        var _this = _super.call(this) || this;
        _this.styleShape = null;
        _this.ab = GeographicPolylineSeries.$;
        return _this;
    }
    GeographicPolylineSeries.prototype.bf = function () {
        return new GeographicPolylineSeriesView(this);
    };
    GeographicPolylineSeries.prototype.iv = function (a) {
        _super.prototype.iv.call(this, a);
        this.mp = a;
    };
    GeographicPolylineSeries.prototype.l2 = function () {
        var a = new ScatterPolylineSeries();
        a.styleShape = delegateCombine(a.styleShape, runOn(this, this.mt));
        return a;
    };
    GeographicPolylineSeries.prototype.mt = function (a, b) {
        if (this.styleShape != null) {
            this.styleShape(this, b);
        }
    };
    GeographicPolylineSeries.prototype.i0 = function (a, b, c, d) {
        _super.prototype.i0.call(this, a, b, c, d);
        switch (b) {
            case GeographicPolylineSeries.$$p[1]:
                this.mp.b8();
                break;
            case GeographicPolylineSeries.$$p[0]:
                this.mp.b9();
                break;
        }
    };
    GeographicPolylineSeries.$t = markType(GeographicPolylineSeries, 'GeographicPolylineSeries', GeographicShapeSeriesBase.$);
    GeographicPolylineSeries.$$p = markDep(DependencyProperty, PropertyMetadata, GeographicPolylineSeries, 'raisePropertyChanged', ['ShapeStyle:mw:mu', [Style.$, null], 'ShapeStyleSelector:mo:mv', [StyleSelector.$, null]]);
    return GeographicPolylineSeries;
}(GeographicShapeSeriesBase));
export { GeographicPolylineSeries };
