// igniteui-react-maps/esm5/lib/GeographicHighDensityScatterSeriesView_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { HostSeriesView } from "./HostSeriesView";
import { GeographicMapSeriesHost } from "./GeographicMapSeriesHost";
import { HighDensityScatterSeries } from "igniteui-react-charts";
import { typeCast, markType, runOn, delegateCombine, markDep } from "igniteui-react-core";
import { DependencyProperty } from "igniteui-react-core";
import { Color } from "igniteui-react-core";
import { PropertyMetadata } from "igniteui-react-core";
/**
 * @hidden
 */
var GeographicHighDensityScatterSeriesView = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicHighDensityScatterSeriesView, _super);
    function GeographicHighDensityScatterSeriesView(a) {
        var _this = _super.call(this, a) || this;
        _this.b0 = null;
        _this.b0 = a;
        return _this;
    }
    GeographicHighDensityScatterSeriesView.prototype.bv = function () {
        _super.prototype.bv.call(this);
        var a = typeCast(HighDensityScatterSeries.$, this.b0.l3);
        a.n4 = this.b0.ms;
        a.n8 = this.b0.mq;
        a.mo = this.b0.mf;
        a.mn = this.b0.me;
        a.c7 = this.b0.md;
        a.ng = this.b0.mh;
        a.nf = this.b0.mg;
    };
    GeographicHighDensityScatterSeriesView.prototype.b6 = function () {
        this.b0.l3.n4 = this.b0.ms;
    };
    GeographicHighDensityScatterSeriesView.prototype.b5 = function () {
        this.b0.l3.n8 = this.b0.mq;
    };
    GeographicHighDensityScatterSeriesView.prototype.cb = function () {
        this.b0.l3.mo = this.b0.mf;
    };
    GeographicHighDensityScatterSeriesView.prototype.b9 = function () {
        this.b0.l3.mn = this.b0.me;
    };
    GeographicHighDensityScatterSeriesView.prototype.b7 = function () {
        this.b0.l3.c7 = this.b0.md;
    };
    GeographicHighDensityScatterSeriesView.prototype.b4 = function () {
        this.b0.l3.ng = this.b0.mh;
    };
    GeographicHighDensityScatterSeriesView.prototype.b2 = function () {
        this.b0.l3.nf = this.b0.mg;
    };
    GeographicHighDensityScatterSeriesView.prototype.ca = function () {
        var a = typeCast(HighDensityScatterSeries.$, this.b0.l3);
        if (this.b0.bs == null) {
            a.l8 = null;
            a.ma = null;
            return;
        }
        a.l8 = this.b0.bs.nc;
        a.ma = this.b0.bs.ne;
    };
    GeographicHighDensityScatterSeriesView.prototype.cc = function () {
        this.b0.l3.l8 = this.b0.bs.nc;
    };
    GeographicHighDensityScatterSeriesView.prototype.cd = function () {
        this.b0.l3.ma = this.b0.bs.ne;
    };
    GeographicHighDensityScatterSeriesView.prototype.b3 = function () {
        this.b0.l3.o1 = this.b0.na;
    };
    GeographicHighDensityScatterSeriesView.prototype.b1 = function () {
        this.b0.l3.o0 = this.b0.m9;
    };
    GeographicHighDensityScatterSeriesView.prototype.b8 = function () {
        this.b0.l3.ns = this.b0.mk;
    };
    GeographicHighDensityScatterSeriesView.$t = markType(GeographicHighDensityScatterSeriesView, 'GeographicHighDensityScatterSeriesView', HostSeriesView.$);
    return GeographicHighDensityScatterSeriesView;
}(HostSeriesView));
export { GeographicHighDensityScatterSeriesView };
/**
 * @hidden
 */
var GeographicHighDensityScatterSeries = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(GeographicHighDensityScatterSeries, _super);
    function GeographicHighDensityScatterSeries() {
        var _this = _super.call(this) || this;
        _this.mb = null;
        _this.progressiveLoadStatusChanged = null;
        _this.mj = 0;
        _this.ab = GeographicHighDensityScatterSeries.$;
        return _this;
    }
    GeographicHighDensityScatterSeries.prototype.bf = function () {
        return new GeographicHighDensityScatterSeriesView(this);
    };
    GeographicHighDensityScatterSeries.prototype.iv = function (a) {
        _super.prototype.iv.call(this, a);
        this.mc = a;
    };
    GeographicHighDensityScatterSeries.prototype.l2 = function () {
        return new HighDensityScatterSeries();
    };
    Object.defineProperty(GeographicHighDensityScatterSeries.prototype, "ml", {
        get: function () {
            return this.mj;
        },
        set: function (a) {
            var b = this.mj;
            this.mj = a;
            this.raisePropertyChanged("ProgressiveStatus", b, this.mj);
        },
        enumerable: true,
        configurable: true
    });
    GeographicHighDensityScatterSeries.prototype.l8 = function () {
        _super.prototype.l8.call(this);
        var a = this.l3;
        a.progressiveLoadStatusChanged = delegateCombine(a.progressiveLoadStatusChanged, runOn(this, this.my));
    };
    GeographicHighDensityScatterSeries.prototype.my = function (a, b) {
        this.ml = b.currentStatus;
        if (this.progressiveLoadStatusChanged != null) {
            this.progressiveLoadStatusChanged(this, b);
        }
    };
    GeographicHighDensityScatterSeries.prototype.i0 = function (a, b, c, d) {
        _super.prototype.i0.call(this, a, b, c, d);
        switch (b) {
            case "SeriesViewer":
                this.mc.ca();
                break;
            case GeographicHighDensityScatterSeries.$$p[5]:
                this.mc.b6();
                break;
            case GeographicHighDensityScatterSeries.$$p[4]:
                this.mc.b5();
                break;
            case GeographicHighDensityScatterSeries.$$p[9]:
                this.mc.cb();
                break;
            case GeographicHighDensityScatterSeries.$$p[8]:
                this.mc.b9();
                break;
            case GeographicHighDensityScatterSeries.$$p[6]:
                this.mc.b7();
                break;
            case "XAxis":
                this.mc.cc();
                break;
            case "YAxis":
                this.mc.cd();
                break;
            case GeographicHighDensityScatterSeries.$$p[3]:
                this.mc.b4();
                break;
            case GeographicHighDensityScatterSeries.$$p[1]:
                this.mc.b2();
                break;
            case GeographicHighDensityScatterSeries.$$p[2]:
                this.mc.b3();
                break;
            case GeographicHighDensityScatterSeries.$$p[0]:
                this.mc.b1();
                break;
            case GeographicHighDensityScatterSeries.$$p[7]:
                this.mc.b8();
                break;
        }
    };
    GeographicHighDensityScatterSeries.prototype.get_dq = function () {
        return true;
    };
    GeographicHighDensityScatterSeries.prototype.set_dq = function (a) {
        _super.prototype.set_dq.call(this, a);
    };
    GeographicHighDensityScatterSeries.$t = markType(GeographicHighDensityScatterSeries, 'GeographicHighDensityScatterSeries', GeographicMapSeriesHost.$);
    GeographicHighDensityScatterSeries.$$p = markDep(DependencyProperty, PropertyMetadata, GeographicHighDensityScatterSeries, 'raisePropertyChanged', ['HeatMaximumColor:m9:mz', [Color.$, null], 'HeatMaximum:mg:m0', [1, 50], 'HeatMinimumColor:na:m1', [Color.$, null], 'HeatMinimum:mh:m2', [1, 0], 'LatitudeMemberPath:mq:m3', [2, null], 'LongitudeMemberPath:ms:m4', [2, null], 'MouseOverEnabled:md:m5', [0, false], 'PointExtent:mk:m6', [1, 1], 'ProgressiveLoad:me:m7', [0, true], 'UseBruteForce:mf:m8', [0, false]]);
    return GeographicHighDensityScatterSeries;
}(GeographicMapSeriesHost));
export { GeographicHighDensityScatterSeries };
