// igniteui-react-maps/esm5/lib/igr-geographic-map-imagery.js
import { delegateCombine, delegateRemove } from "igniteui-react-core";
import { IgrImageTilesReadyEventArgs } from "./igr-image-tiles-ready-event-args";
import { IgrImagesChangedEventArgs } from "./igr-images-changed-event-args";
import { IgrCancellingMultiScaleImageEventArgs } from "igniteui-react-core";
import { IgrDownloadingMultiScaleImageEventArgs } from "igniteui-react-core";
import { fromRect, toRect } from "igniteui-react-core";
/**
 * Control class used for displaying multi-scale TileSource imagery.  Designed to be used for background imagery in the IgxGeographicMap.Component
*/
var IgrGeographicMapImagery = /** @class */ /*@__PURE__*/ (function () {
    function IgrGeographicMapImagery() {
        this._imageTilesReady = null;
        this._imageTilesReady_wrapped = null;
        this._imagesChanged = null;
        this._imagesChanged_wrapped = null;
        this._cancellingImage = null;
        this._cancellingImage_wrapped = null;
        this._downloadingImage = null;
        this._downloadingImage_wrapped = null;
        this._implementation = this.createImplementation();
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    IgrGeographicMapImagery.prototype.createImplementation = function () {
        return null;
    };
    Object.defineProperty(IgrGeographicMapImagery.prototype, "i", {
        /**
         * @hidden
         */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    IgrGeographicMapImagery.prototype.onImplementationCreated = function () {
    };
    IgrGeographicMapImagery.prototype._provideImplementation = function (i) {
        this._implementation = i;
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    };
    Object.defineProperty(IgrGeographicMapImagery.prototype, "deferralHandler", {
        /**
         * The deferral handler to use for deferred refreshes.
        */
        get: function () {
            return this.i.deferralHandler;
        },
        set: function (v) {
            this.i.deferralHandler = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "windowRect", {
        /**
         * A rectangle representing the portion of the map imagery currently in view.
         * A rectangle at X=0, Y=0 with a Height and Width of 1 implies the entire plotting area is in view.  A Height and Width of .5 would imply that the view is halfway zoomed in.
        */
        get: function () {
            return fromRect(this.i.windowRect);
        },
        set: function (v) {
            this.i.windowRect = toRect(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "geographicMap", {
        /**
         * The IgxGeographicMapComponent which is hosting this GeographicMapImagery control.
        */
        get: function () {
            if (this.i.geographicMap == null) {
                return null;
            }
            return this.i.geographicMap.externalObject;
        },
        set: function (v) {
            v == null ? this.i.geographicMap = null : this.i.geographicMap = v.i;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "referer", {
        /**
         * Gets or sets the referer sent in the headers of the tile request.
        */
        get: function () {
            return this.i.referer;
        },
        set: function (v) {
            this.i.referer = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "userAgent", {
        /**
         * Gets or sets the UserAgent sent in the headers of the tile request.
        */
        get: function () {
            return this.i.userAgent;
        },
        set: function (v) {
            this.i.userAgent = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "opacity", {
        get: function () {
            return this.i.opacity;
        },
        set: function (v) {
            this.i.opacity = +v;
        },
        enumerable: true,
        configurable: true
    });
    IgrGeographicMapImagery.prototype.findByName = function (name) {
        if (this.geographicMap && this.geographicMap.name && this.geographicMap.name == name) {
            return this.geographicMap;
        }
        return null;
    };
    /**
     * Removes all cached tile Imagery from the
    
    */
    IgrGeographicMapImagery.prototype.clearTileCache = function () {
        this.i.clearTileCache();
    };
    Object.defineProperty(IgrGeographicMapImagery.prototype, "imageTilesReady", {
        /**
         * Event raised when the image tiles have been downloaded.
        */
        get: function () {
            return this._imageTilesReady;
        },
        set: function (ev) {
            var _this = this;
            if (this._imageTilesReady_wrapped !== null) {
                this.i.imageTilesReady = delegateRemove(this.i.imageTilesReady, this._imageTilesReady_wrapped);
                this._imageTilesReady_wrapped = null;
                this._imageTilesReady = null;
            }
            this._imageTilesReady = ev;
            this._imageTilesReady_wrapped = function (o, e) {
                var outerArgs = new IgrImageTilesReadyEventArgs();
                outerArgs._provideImplementation(e);
                if (_this.beforeImageTilesReady) {
                    _this.beforeImageTilesReady(_this, outerArgs);
                }
                if (_this._imageTilesReady) {
                    _this._imageTilesReady(_this, outerArgs);
                }
            };
            this.i.imageTilesReady = delegateCombine(this.i.imageTilesReady, this._imageTilesReady_wrapped);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "imagesChanged", {
        /**
         * Event raised when tile images have changed.
        */
        get: function () {
            return this._imagesChanged;
        },
        set: function (ev) {
            var _this = this;
            if (this._imagesChanged_wrapped !== null) {
                this.i.imagesChanged = delegateRemove(this.i.imagesChanged, this._imagesChanged_wrapped);
                this._imagesChanged_wrapped = null;
                this._imagesChanged = null;
            }
            this._imagesChanged = ev;
            this._imagesChanged_wrapped = function (o, e) {
                var outerArgs = new IgrImagesChangedEventArgs();
                outerArgs._provideImplementation(e);
                if (_this.beforeImagesChanged) {
                    _this.beforeImagesChanged(_this, outerArgs);
                }
                if (_this._imagesChanged) {
                    _this._imagesChanged(_this, outerArgs);
                }
            };
            this.i.imagesChanged = delegateCombine(this.i.imagesChanged, this._imagesChanged_wrapped);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "cancellingImage", {
        /**
         * Event raised when an image is being downloaded.
        */
        get: function () {
            return this._cancellingImage;
        },
        set: function (ev) {
            var _this = this;
            if (this._cancellingImage_wrapped !== null) {
                this.i.cancellingImage = delegateRemove(this.i.cancellingImage, this._cancellingImage_wrapped);
                this._cancellingImage_wrapped = null;
                this._cancellingImage = null;
            }
            this._cancellingImage = ev;
            this._cancellingImage_wrapped = function (o, e) {
                var outerArgs = new IgrCancellingMultiScaleImageEventArgs();
                outerArgs._provideImplementation(e);
                if (_this.beforeCancellingImage) {
                    _this.beforeCancellingImage(_this, outerArgs);
                }
                if (_this._cancellingImage) {
                    _this._cancellingImage(_this, outerArgs);
                }
            };
            this.i.cancellingImage = delegateCombine(this.i.cancellingImage, this._cancellingImage_wrapped);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicMapImagery.prototype, "downloadingImage", {
        /**
         * Event raised when an image is being downloaded.
        */
        get: function () {
            return this._downloadingImage;
        },
        set: function (ev) {
            var _this = this;
            if (this._downloadingImage_wrapped !== null) {
                this.i.downloadingImage = delegateRemove(this.i.downloadingImage, this._downloadingImage_wrapped);
                this._downloadingImage_wrapped = null;
                this._downloadingImage = null;
            }
            this._downloadingImage = ev;
            this._downloadingImage_wrapped = function (o, e) {
                var outerArgs = new IgrDownloadingMultiScaleImageEventArgs();
                outerArgs._provideImplementation(e);
                if (_this.beforeDownloadingImage) {
                    _this.beforeDownloadingImage(_this, outerArgs);
                }
                if (_this._downloadingImage) {
                    _this._downloadingImage(_this, outerArgs);
                }
            };
            this.i.downloadingImage = delegateCombine(this.i.downloadingImage, this._downloadingImage_wrapped);
        },
        enumerable: true,
        configurable: true
    });
    return IgrGeographicMapImagery;
}());
export { IgrGeographicMapImagery };
