// igniteui-react-maps/esm5/lib/igr-geographic-shape-series-base.js
import * as tslib_1 from "tslib";
import { IgrGeographicMapSeriesHost } from "./igr-geographic-map-series-host";
import { IgrShapeDataSource } from "igniteui-react-core";
import { delegateCombine, delegateRemove } from "igniteui-react-core";
/**
 * Base class for series which render polygons on a map.
*/
var IgrGeographicShapeSeriesBase = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(IgrGeographicShapeSeriesBase, _super);
    function IgrGeographicShapeSeriesBase(props) {
        var _this = _super.call(this, props) || this;
        _this._currData = null;
        _this._onImportCompleted = null;
        return _this;
    }
    Object.defineProperty(IgrGeographicShapeSeriesBase.prototype, "i", {
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicShapeSeriesBase.prototype, "shapeDataSource", {
        get: function () {
            return this._shapeDataSource;
        },
        /**
         * Gets or sets the string path of the .shp portion of the Shapefile.
        */
        set: function (value) {
            this._shapeDataSource = value;
            this.updateShapeDataSource();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicShapeSeriesBase.prototype, "databaseSource", {
        get: function () {
            return this._databaseSource;
        },
        /**
         * Gets or sets the string path of the .dbf portion of the Shapefile.
        */
        set: function (value) {
            this._databaseSource = value;
            this.updateShapeDataSource();
        },
        enumerable: true,
        configurable: true
    });
    IgrGeographicShapeSeriesBase.prototype.onUpdateDataSource = function (dataSource) {
        var _this = this;
        if (this._onImportCompleted == null) {
            this._onImportCompleted = function (s, e) {
                if (_this._currData) {
                    _this._currData.i.importCompleted = delegateRemove(_this._currData.i.importCompleted, _this._onImportCompleted);
                    var data = _this._currData;
                    _this._currData = null;
                    _this.i.itemsSource = data.getPointData();
                }
            };
        }
        if (dataSource && dataSource.getPointData) {
            if (dataSource.count > 0) {
                return dataSource.getPointData();
            }
            this._currData = dataSource;
            this._currData.i.importCompleted = delegateCombine(this._currData.i.importCompleted, this._onImportCompleted);
            return null;
        }
        return dataSource;
    };
    IgrGeographicShapeSeriesBase.prototype.updateShapeDataSource = function () {
        if (this._cachedShapeDataSource == null) {
            this._cachedShapeDataSource = new IgrShapeDataSource();
            this.onShapeDataSourceImportCompleted = this.onShapeDataSourceImportCompleted.bind(this);
            this._cachedShapeDataSource.importCompleted = this.onShapeDataSourceImportCompleted;
        }
        this._cachedShapeDataSource.shapefileSource = this._shapeDataSource;
        this._cachedShapeDataSource.databaseSource = this._databaseSource;
        if (this._shapeDataSource && this._databaseSource) {
            this._cachedShapeDataSource.dataBind();
        }
    };
    IgrGeographicShapeSeriesBase.prototype.onShapeDataSourceImportCompleted = function (s, e) {
        this.dataSource = this._cachedShapeDataSource.getPointData();
    };
    Object.defineProperty(IgrGeographicShapeSeriesBase.prototype, "shapeMemberPath", {
        /**
     * The name of the property on ItemsSource items which, for each shape, contains a list of points to be converted to a polygon.
     * To be consistent with the Shapefile technical description, it is expected that each list of points is defined as an IEnumerable of IEnumerable of Point, or in other words, a list of lists of points.
    */
        get: function () {
            return this.i.mi;
        },
        set: function (v) {
            this.i.mi = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicShapeSeriesBase.prototype, "shapefileDataSource", {
        /**
         * Set a shapefile datasource to use with the series.
        */
        get: function () {
            if (this.i.md == null) {
                return null;
            }
            if (!this.i.md.externalObject) {
                var e = new IgrShapeDataSource();
                e._implementation = this.i.md;
                this.i.md.externalObject = e;
            }
            return this.i.md.externalObject;
        },
        set: function (v) {
            v == null ? this.i.md = null : this.i.md = v.i;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgrGeographicShapeSeriesBase.prototype, "shapeFilterResolution", {
        /**
         * The resolution at which to filter out shapes in the series.  For example, if the ShapeFilterResolution is set to 3, then elements with a bounding rectangle smaller than 3 X 3 pixels will be filtered out.
         * In the case of GeographicPolylineSeries, the resolution is compared to either dimension, rather than both.  In other words, a polyline will not be filtered if its height or its width exceeds the value of this property.  Whereas with a ShapeSeries, both the height and the width must exceed the value of this property.
        */
        get: function () {
            return this.i.mf;
        },
        set: function (v) {
            this.i.mf = +v;
        },
        enumerable: true,
        configurable: true
    });
    IgrGeographicShapeSeriesBase.prototype.findByName = function (name) {
        var baseResult = _super.prototype.findByName.call(this, name);
        if (baseResult) {
            return baseResult;
        }
        if (this.shapefileDataSource && this.shapefileDataSource.name && this.shapefileDataSource.name == name) {
            return this.shapefileDataSource;
        }
        return null;
    };
    return IgrGeographicShapeSeriesBase;
}(IgrGeographicMapSeriesHost));
export { IgrGeographicShapeSeriesBase };
