// igniteui-react-maps/esm5/lib/SphericalMercatorVerticalScaler.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import * as tslib_1 from "tslib";
import { VerticalLinearScaler } from "igniteui-react-charts";
import { Rect } from "igniteui-react-core";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var SphericalMercatorVerticalScaler = /** @class */ /*@__PURE__*/ (function (_super) {
    tslib_1.__extends(SphericalMercatorVerticalScaler, _super);
    function SphericalMercatorVerticalScaler() {
        var _this = _super.call(this) || this;
        _this.au = null;
        _this.ai = 0;
        _this.aj = 0;
        _this.au = new Rect(0, 0, 0, 1, 1);
        return _this;
    }
    SphericalMercatorVerticalScaler.prototype.q = function (a, b) {
        if (!b.c.isEmpty) {
            var c = a * b.e.height + b.e.top * b.d.height;
            var d = this.getMercatorUnscaledValue(c, this.au, b.c, b.a);
            return d;
        }
        else {
            return this.getMercatorUnscaledValue(a, b.e, b.d, b.a);
        }
    };
    SphericalMercatorVerticalScaler.prototype.p = function (a, b) {
        if (!b.c.isEmpty) {
            var c = this.getMercatorScaledValue1(a, this.au, b.c, b.a);
            c = (c - (b.e.top * b.d.height)) / b.e.height;
            return c;
        }
        else {
            return this.getMercatorScaledValue1(a, b.e, b.d, b.a);
        }
    };
    SphericalMercatorVerticalScaler.prototype.af = function (values_) {
        var a = (Array.isArray(values_) ? values_ : null);
        return a;
        return null;
    };
    SphericalMercatorVerticalScaler.prototype.v = function (a, b, c, d) {
        var e;
        var f = d.e;
        var g = d.d;
        var h = d.c;
        var i = d.a;
        var j = !h.isEmpty;
        var k = this.m;
        var l = this.o;
        var m = h.top;
        var n = h.height;
        var o = f.top;
        var p = f.height;
        var q = g.top;
        var r = g.height;
        var s = this.au.top;
        var t = this.au.height;
        var u = this.af(a);
        var v = 0.017453292519943295;
        var w;
        var x;
        var y;
        var z;
        var aa;
        var ab = false;
        if (u != null) {
            ab = true;
        }
        for (var ac = b; ac < c; ac++) {
            if (ab) {
                e = u[ac];
            }
            else {
                e = a.item(ac);
            }
            if (j) {
                if (e < SphericalMercatorVerticalScaler.ah) {
                    e = SphericalMercatorVerticalScaler.ah;
                }
                if (e > SphericalMercatorVerticalScaler.ag) {
                    e = SphericalMercatorVerticalScaler.ag;
                }
                w = e * (v);
                x = Math.sin(w);
                y = 0.5 * Math.log((1 + x) / (1 - x));
                z = (this.ai - y) * n / this.aj;
                z = m + z;
                aa = (z - s * n) / t;
                z = (aa - (o * r)) / p;
                if (ab) {
                    u[ac] = z;
                }
                else {
                    a.item(ac, z);
                }
            }
            else {
                if (e < SphericalMercatorVerticalScaler.ah) {
                    e = SphericalMercatorVerticalScaler.ah;
                }
                if (e > SphericalMercatorVerticalScaler.ag) {
                    e = SphericalMercatorVerticalScaler.ag;
                }
                w = e * (v);
                x = Math.sin(w);
                y = 0.5 * Math.log((1 + x) / (1 - x));
                z = (this.ai - y) * r / this.aj;
                z = q + z;
                z = (z - o * r) / p;
                if (ab) {
                    u[ac] = z;
                }
                else {
                    a.item(ac, z);
                }
            }
        }
    };
    SphericalMercatorVerticalScaler.prototype.t = function (a, b, c, d, e) {
        d = this.ak(a.kj);
        e = this.ak(a.ki);
        return {
            p3: d,
            p4: e
        };
    };
    SphericalMercatorVerticalScaler.prototype.w = function (a, b, c) {
        _super.prototype.w.call(this, a, b, c);
        switch (a) {
            case "ActualMinimumValue":
            case "ActualMaximumValue":
                this.ai = SphericalMercatorVerticalScaler.am(this.n);
                this.aj = SphericalMercatorVerticalScaler.am(this.n) - SphericalMercatorVerticalScaler.am(this.o);
                break;
        }
    };
    Object.defineProperty(SphericalMercatorVerticalScaler, "as", {
        get: function () {
            return SphericalMercatorVerticalScaler.ag;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SphericalMercatorVerticalScaler, "at", {
        get: function () {
            return SphericalMercatorVerticalScaler.ah;
        },
        enumerable: true,
        configurable: true
    });
    SphericalMercatorVerticalScaler.prototype.getMercatorUnscaledValue = function (a, b, c, d) {
        var e = (a - c.top) * b.width + b.top * c.height;
        e = SphericalMercatorVerticalScaler.am(this.n) - e / (c.height / this.aj);
        return SphericalMercatorVerticalScaler.aq(e);
    };
    SphericalMercatorVerticalScaler.prototype.getMercatorScaledValue1 = function (a, b, c, d) {
        var e = c.top + this.getMercatorScaledValue3(a, c.height);
        return (e - b.top * c.height) / b.width;
    };
    SphericalMercatorVerticalScaler.prototype.getMercatorScaledValue2 = function (a, b, c, d, e) {
        var f = this.getMercatorScaledValue3(a, d.height);
        return (f - b.top * c.height) / b.width;
    };
    SphericalMercatorVerticalScaler.prototype.getMercatorScaledValue3 = function (a, b) {
        a = a > SphericalMercatorVerticalScaler.ah ? (a < SphericalMercatorVerticalScaler.ag ? a : SphericalMercatorVerticalScaler.ag) : SphericalMercatorVerticalScaler.ah;
        var c = SphericalMercatorVerticalScaler.am(a);
        var d = (this.ai - c) * b / this.aj;
        return (d);
    };
    SphericalMercatorVerticalScaler.am = function (a) {
        var b = a * (0.017453292519943295);
        var c = Math.sin(b);
        var d = 0.5 * Math.log((1 + c) / (1 - c));
        return d;
    };
    SphericalMercatorVerticalScaler.aq = function (a) {
        a = Math.exp(2 * a);
        return Math.asin((a - 1) / (a + 1)) / 0.017453292519943295;
    };
    SphericalMercatorVerticalScaler.prototype.ak = function (a) {
        if (a < SphericalMercatorVerticalScaler.ah) {
            return SphericalMercatorVerticalScaler.ah;
        }
        if (a > SphericalMercatorVerticalScaler.ag) {
            return SphericalMercatorVerticalScaler.ag;
        }
        return a;
    };
    SphericalMercatorVerticalScaler.$t = markType(SphericalMercatorVerticalScaler, 'SphericalMercatorVerticalScaler', VerticalLinearScaler.$);
    SphericalMercatorVerticalScaler.ag = 85.05112878;
    SphericalMercatorVerticalScaler.ah = -85.05112878;
    return SphericalMercatorVerticalScaler;
}(VerticalLinearScaler));
export { SphericalMercatorVerticalScaler };
