// igniteui-react-maps/esm2015/lib/BingMapsImageryStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Describes available map styles for the Bing Maps imagery.
 */
export var BingMapsImageryStyle = /*@__PURE__*/ (function (BingMapsImageryStyle) {
    /**
     * Specifies the Aerial map style without road or labels overlay.
     */
    BingMapsImageryStyle[BingMapsImageryStyle["Aerial"] = 0] = "Aerial";
    /**
     * Specifies the Aerial map style with road and labels overlay.
     */
    BingMapsImageryStyle[BingMapsImageryStyle["AerialWithLabels"] = 1] = "AerialWithLabels";
    /**
     * Specifies the Roads map style without aerial overlay.
     */
    BingMapsImageryStyle[BingMapsImageryStyle["Road"] = 2] = "Road";
    /**
     * Specifies a dark version of the Roads maps.
     */
    BingMapsImageryStyle[BingMapsImageryStyle["CanvasDark"] = 3] = "CanvasDark";
    /**
     * Specifies a lighter version of the Roads maps.
     */
    BingMapsImageryStyle[BingMapsImageryStyle["CanvasLight"] = 4] = "CanvasLight";
    /**
     * Specifies a grayscale version of the Roads maps.
     */
    BingMapsImageryStyle[BingMapsImageryStyle["CanvasGray"] = 5] = "CanvasGray";
    return BingMapsImageryStyle;
})({});
/**
 * @hidden
 */
export let BingMapsImageryStyle_$type = /*@__PURE__*/ markEnum('BingMapsImageryStyle', 'Aerial,0|AerialWithLabels,1|Road,2|CanvasDark,3|CanvasLight,4|CanvasGray,5');
