// igniteui-react-maps/esm2015/lib/igr-geographic-xy-triangulating-series.js
import { IgrGeographicMapSeriesHost } from "./igr-geographic-map-series-host";
import { IgrTriangulationDataSource } from "igniteui-react-core";
/**
 * Base class for geographic series which triangulate XY data prior to rendering.
*/
export class IgrGeographicXYTriangulatingSeries extends IgrGeographicMapSeriesHost {
    constructor(props) {
        super(props);
    }
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the string path specifying the location of the Itf file.
    */
    set triangulationDataSource(value) {
        this._triangulationDataSource = value;
        this.updateTriangulationDataSource();
    }
    get triangulationDataSource() {
        return this._triangulationDataSource;
    }
    updateTriangulationDataSource() {
        if (this._cachedTriangulationDataSource == null) {
            this._cachedTriangulationDataSource = new IgrTriangulationDataSource();
            this.onTriangulationDataSourceImportCompleted = this.onTriangulationDataSourceImportCompleted.bind(this);
            this._cachedTriangulationDataSource.importCompleted = this.onTriangulationDataSourceImportCompleted;
        }
        this._cachedTriangulationDataSource.source = this._triangulationDataSource;
    }
    onTriangulationDataSourceImportCompleted(s, e) {
        this.dataSource = this._cachedTriangulationDataSource.getPointData();
        this.trianglesSource = this._cachedTriangulationDataSource.getTriangleData();
    }
    /**
 * The name of the property from which to extract the Longitude for each item in the ItemsSource.
*/
    get longitudeMemberPath() {
        return this.i.mf;
    }
    set longitudeMemberPath(v) {
        this.i.mf = v;
    }
    /**
     * The name of the property from which to extract the Latitude for each item in the ItemsSource.
    */
    get latitudeMemberPath() {
        return this.i.md;
    }
    set latitudeMemberPath(v) {
        this.i.md = v;
    }
    /**
     * The source of triangulation data.
     * This property is optional.  If it is left as null, the triangulation will be created based on the items in the ItemsSource.  Triangulation is a demanding operation, so the runtime performance will be better when specifying a TriangulationSource, especially when a large number of data items are present.
    */
    get trianglesSource() {
        return this.i.mc;
    }
    set trianglesSource(v) {
        this.i.mc = v;
    }
    /**
     * The name of the property of the TrianglesSource items which, for each triangle, contains the index of the first vertex point in the ItemsSource.
    */
    get triangleVertexMemberPath1() {
        return this.i.mi;
    }
    set triangleVertexMemberPath1(v) {
        this.i.mi = v;
    }
    /**
     * The name of the property of the TrianglesSource items which, for each triangle, contains the index of the second vertex point in the ItemsSource.
    */
    get triangleVertexMemberPath2() {
        return this.i.mk;
    }
    set triangleVertexMemberPath2(v) {
        this.i.mk = v;
    }
    /**
     * The name of the property of the TrianglesSource items which, for each triangle, contains the index of the third vertex point in the ItemsSource.
    */
    get triangleVertexMemberPath3() {
        return this.i.mm;
    }
    set triangleVertexMemberPath3(v) {
        this.i.mm = v;
    }
}
