// igniteui-react-maps/esm2015/lib/igr-tile-series.js
import { IgrShapeSeriesBase } from "igniteui-react-charts";
import { TileSeries } from "./TileSeries";
/**
 * Represents a tile series class that can render imagery tiles
*/
export class IgrTileSeries extends IgrShapeSeriesBase {
    createImplementation() {
        return new TileSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    constructor(props) {
        super(props);
    }
    /**
     * The MapImagery which is used to provide the tiles for display.
    */
    get tileImagery() {
        if (this.i.ng == null) {
            return null;
        }
        return this.i.ng.externalObject;
    }
    set tileImagery(v) {
        v == null ? this.i.ng = null : this.i.ng = v.i;
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.tileImagery && this.tileImagery.name && this.tileImagery.name == name) {
            return this.tileImagery;
        }
        return null;
    }
    /**
     * Defers rendering of the series
    
    */
    deferredRefresh() {
        this.i.deferredRefresh();
    }
}
