// igniteui-react-maps/esm2015/lib/igr-arc-gis-online-map-imagery.js
import { IgrGeographicMapImagery } from "./igr-geographic-map-imagery";
import { ArcGISOnlineMapImagery as ArcGISOnlineMapImagery_internal } from "./ArcGISOnlineMapImagery";
import { ensureBool } from "igniteui-react-core";
/**
 * UI element class used to display ArcGIS Online Map imagery in the IgxGeographicMap.Component
*/
export class IgrArcGISOnlineMapImagery extends IgrGeographicMapImagery {
    createImplementation() {
        return new ArcGISOnlineMapImagery_internal();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    constructor() {
        super();
    }
    /**
     * Gets or sets the map tile server uri.
    */
    get mapServerUri() {
        return this.i.mapServerUri;
    }
    set mapServerUri(v) {
        this.i.mapServerUri = v;
    }
    /**
     * Gets or sets the UserName.
    */
    get userName() {
        return this.i.userName;
    }
    set userName(v) {
        this.i.userName = v;
    }
    /**
     * Gets or sets the Password.
    */
    get password() {
        return this.i.password;
    }
    set password(v) {
        this.i.password = v;
    }
    /**
     * Gets or sets the UserToken.
    */
    get userToken() {
        return this.i.userToken;
    }
    set userToken(v) {
        this.i.userToken = v;
    }
    /**
     * Gets or sets the IsMapPublic variable.  A value of false requires authentication parameters.
    */
    get isMapPublic() {
        return this.i.isMapPublic;
    }
    set isMapPublic(v) {
        this.i.isMapPublic = ensureBool(v);
    }
    /**
     * Gets or Sets the Default Token Timeout before the Token Expires, this value is in Minutes
    */
    get defaultTokenTimeout() {
        return this.i.defaultTokenTimeout;
    }
    set defaultTokenTimeout(v) {
        this.i.defaultTokenTimeout = +v;
    }
    /**
     * Gets or Sets the current endpoint to retrieve the token.
    */
    get tokenGenerationEndPoint() {
        return this.i.tokenGenerationEndPoint;
    }
    set tokenGenerationEndPoint(v) {
        this.i.tokenGenerationEndPoint = v;
    }
    /**
     * Contains the User Supplied Referer Uri to be used in Token Generation.
    */
    get refererUri() {
        return this.i.refererUri;
    }
    set refererUri(v) {
        this.i.refererUri = v;
    }
    /**
     * Retreives a new token based on the preconfigured parameters and updates the UserToken field, Normal this is called automatically.
     * The is provided to the end-user to allow them to reteive tokens on their onw.
    
    */
    acquireNewToken() {
        this.i.acquireNewToken();
    }
}
